examples/CMakeLists.txt

```diff
@@ -42,130 +42,37 @@ project(examples)
-# are going to compile one of the dlib example programs which has only one .cpp
-# file, assignment_learning_ex.cpp.  If your program consisted of multiple .cpp
-# files you would simply list them here in the add_executable() statement.  
-add_executable(assignment_learning_ex assignment_learning_ex.cpp)
-# Finally, you need to tell CMake that this program, assignment_learning_ex,
-# depends on dlib.  You do that with this statement: 
-target_link_libraries(assignment_learning_ex dlib::dlib)
-
-
-
-# To compile this program all you need to do is ask cmake.  You would type
-# these commands from within the directory containing this CMakeLists.txt
-# file:
-#   mkdir build
-#   cd build
-#   cmake ..
-#   cmake --build . --config Release
-#
-# The cmake .. command looks in the parent folder for a file named
-# CMakeLists.txt, reads it, and sets up everything needed to build program.
-# Also, note that CMake can generate Visual Studio or XCode project files.  So
-# if instead you had written:
-#   cd build
-#   cmake .. -G Xcode
-#
-# You would be able to open the resulting Xcode project and compile and edit
-# the example programs within the Xcode IDE.  CMake can generate a lot of
-# different types of IDE projects.  Run the cmake -h command to see a list of
-# arguments to -G to see what kinds of projects cmake can generate for you.  It
-# probably includes your favorite IDE in the list.
-
-
-
-
-#################################################################################
-#################################################################################
-#  A CMakeLists.txt file can compile more than just one program.  So below we
-#  tell it to compile the other dlib example programs using pretty much the
-#  same CMake commands we used above.
-#################################################################################
-#################################################################################
-
-
-# Since there are a lot of examples I'm going to use a macro to simplify this
-# CMakeLists.txt file.  However, usually you will create only one executable in
-# your cmake projects and use the syntax shown above.
 macro(add_example name)
    add_executable(${name} ${name}.cpp)
-   target_link_libraries(${name} dlib::dlib )
-
-   if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.13.0") 
-   # And strip symbols to make your binary smaller if you like.  Certainly not
-   # required though.
-       target_link_options(${name} PRIVATE $<$<CONFIG:RELEASE>:-s>)
-   endif()
-endmacro()
-
-# if an example requires GUI, call this macro to check DLIB_NO_GUI_SUPPORT to include or exclude
-macro(add_gui_example name)
-   if (DLIB_NO_GUI_SUPPORT)
-      message("No GUI support, so we won't build the ${name} example.")
-   else()
-      add_example(${name})
-   endif()
+   target_link_libraries(${name} dlib::dlib)
+   target_link_options(${name} PRIVATE $<$<AND:$<CONFIG:RELEASE>,${IS_NOT_WIN32}>:-s>)
```

Comment:
What's this for?

Reply:
This is stripping binaries on Linux. -s isn't a valid option with MSVC

Reply:
Generator expressions are cool. They don't have the nicest syntax but they are better that a rat's nest of if-else statements.

Reply:
Right, but I mean, add a comment saying like "In cmake you can optionally say stuff like this, which directs cmake to apply the -s compile option in release builds in linux, making your binaries smaller.".  Otherwise someone coming to this will just see magic and wonder if it's needed and get nervous about it.  Then there will end up being blog posts about "how to do dlib" that will say "oh you have to do -s or it doesn't work".  And invariably that will be applied in windows and then visual studio will error about it and people will email me about it.  It just goes and goes :) 

Having clear docs and examples and all that makes a big difference.